compiler/rustc_hir_analysis/src/check/region.rs

```diff
@@ -168,9 +168,7 @@ fn resolve_block<'tcx>(visitor: &mut RegionResolutionVisitor<'tcx>, blk: &'tcx h
             }
         }
         if let Some(tail_expr) = blk.expr {
-            if visitor.tcx.features().shorter_tail_lifetimes
-                && blk.span.edition().at_least_rust_2024()
-            {
+            if blk.span.edition().at_least_rust_2021() {
```

Comment:
Question: note that this will limit your analysis of breakage to crates that are `edition=2021` and beyond. Does previous editions have tail expr drop changes?

Reply:
Forgot to say. I think not and the drop order should have been consistent before and including Edition 2021.